test/e2e/e2e_test.go

```diff
@@ -51,6 +51,36 @@ var _ = Describe("Workload cluster creation", func() {
 			By("Creating a high available cluster")
 			if upgradeTest {
 				upgradeManagementCluster()
+			} else if ephemeralTest {
```

Comment:
It would be nice to avoid duplicating the cluster creation here I think. What if we change the logic like this?

```go
if upgradeTest {
    upgradeManagementCluster()
} else {
    // ApplyClusterTemplateAndWait... same for both ephemeral and "normal" tests
    if ephemeralTest {
        // Run tests in ephemeral cluster
    } else {
        // Run tests with pivot
    }
}
```

Reply:
Yes I am working on it.